File: samples/Caliburn.Micro.ViewFirst/Caliburn.Micro.ViewFirst/ShellViewModel.cs
namespace Caliburn.Micro.ViewFirst {
    using System.ComponentModel.Composition;
    using System.Windows;

    [Export("Shell", typeof(IShell))]
    public class ShellViewModel : PropertyChangedBase, IShell {
        string name;

        public string Name {
            get { return name; }
            set {
                name = value;
                NotifyOfPropertyChange(() => Name);
                NotifyOfPropertyChange(() => CanSayHello);
            }
        }

        public bool CanSayHello {
            get { return !string.IsNullOrWhiteSpace(Name); }
        }

        public void SayHello() {
            MessageBox.Show(string.Format("Hello {0}!", Name));
        }
    }
}

File: samples/scenarios/Scenario.Functional.Core/ShellViewModel.fs
namespace Scenario.Functional.Core.ViewModels

open Caliburn.Micro

type ShellViewModel() = 
    inherit Screen()

    let mutable name = "Enter your name"

    override this.OnInitialize () =
        this.DisplayName <- "Shell"

    member this.Name
        with get() = name
        and set(value) =
            name <- value
            base.NotifyOfPropertyChange()

    member this.SayHello () =
        this.DisplayName <- "Hello " + this.Name